usr.sbin/bsdinstall/scripts/zfsboot

```diff
@@ -841,6 +841,18 @@ zfs_create_diskpart()
 
 	case "$ZFSBOOT_PARTITION_SCHEME" in
 	""|GPT*) f_dprintf "$funcname: Creating GPT layout..."
+		#
+		# 0. Check for potential GPT label conflicts with disks not
+		# managed by us and adjust $index accordingly
+		#
+		while [ -e /dev/gpt/efiboot$index ] ||
+			[ -e /dev/gpt/gptboot$index ] ||
```

Comment:
This needs to likely be in the caller so that we catch it for UFS installs as well.
Also, there needs to be a limit so we don't loop forever.

Reply:
Thank you for the review!

AFAICT, the UFS installation code does not generate any GPT labels (other than the ones explicitly set by user) and does try to use them to refer to partitions (/dev/gpt/...) (where this bug surfaces). So here's the only place where the problem happens.

As for setting a limit for the loop, probably you can't have an indefinite number of GPT labels, so IMHO the loop will definitely finish. Still, for the sake of readability, a hard limit could be added. What about 4096? Googling the drive count limits on FreeBSD, I've found a report of a kernel panic with more than 4096 drives: https://forums.FreeBSD.org/threads/maximum-number-of-disks.87168/post-588524 , so we could use it as guidance ;-)

Reply:
OK. That makes sense... I'll admit I didn't go look, but relied on my memory that the ufs path does this too.

4096 is likely a good limit. 10 likely is a good limit too, but you never know who might have huge 'imaging farms' or something.  There's no real hard limit, but I'd like some protection from the installer hanging for reasons that are opaque to the user.

Reply:
OK. That makes sense... I'll admit I didn't go look, but relied on my memory that the ufs path does this too.

4096 is likely a good limit. 10 likely is a good limit too, but you never know who might have huge 'imaging farms' or something.  There's no real hard limit, but I'd like some protection from the installer hanging for reasons that are opaque to the user.

Reply:
Added a limit.